pkg/operator/controller/externaldns/deployment_test.go

```diff
@@ -3081,6 +3091,9 @@ func TestEnsureExternalDNSDeployment(t *testing.T) {
 							"app.kubernetes.io/name":     ExternalDNSBaseName,
 						},
 					},
+					Strategy: appsv1.DeploymentStrategy{
```

Comment:
Why do you need to add the `DeploymentStrategy` to all of these tests?  Isn't is defaulted in `desiredExternalDNSDeployment`?

Reply:
Each testcase in this test function sets `expectedDeployment` which is how the deployment should look like after `ensureExternalDNSDeployment` returns.
This test file needs to be refactored to remove the boilerplate. There was already some intent in that direction like this [externaldns CR builder](https://github.com/openshift/external-dns-operator/blob/main/pkg/operator/controller/utils/test/builders.go#L25). Similar builder needs to be done for the deployment.